sdk/include/opentelemetry/sdk/metrics/view/predicate.h

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include <vector>
+#  if (__GNUC__ == 4 && (__GNUC_MINOR__ == 8 || __GNUC_MINOR__ == 9))
+#    define HAVE_WORKING_REGEX 0
+#  else
+#    include <regex>
+#    define HAVE_WORKING_REGEX 1
+#  endif
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+#  include "opentelemetry/nostd/string_view.h"
+namespace sdk
+{
+namespace metrics
+{
+class Predicate
+{
+public:
+  virtual ~Predicate()                                                        = default;
+  virtual bool Match(opentelemetry::nostd::string_view string) const noexcept = 0;
+};
+
+class PatternPredicate : public Predicate
+{
+public:
+  PatternPredicate(opentelemetry::nostd::string_view pattern) : reg_key_{pattern.data()}
+  {
+    if (pattern == "*")
```

Comment:
I don't think you need to support `*` in all your predicates.  You could have ONE predicate which is the "star" predicate and then a "predicate factory" that returns either a new `PatternPredicate` or `MatchEverything` predicate, if you wanted.

Only commenting because I noticed this logic duplicate in the "ExactPredicate" code as well.

Reply:
Good point. Will change accordingly.

Reply:
This is changed now.